src/main/java/org/apache/ibatis/session/Configuration.java

```diff
@@ -393,7 +393,7 @@ public ExecutorType getDefaultExecutorType() {
   }
 
   public void setDefaultExecutorType(ExecutorType defaultExecutorType) {
-    this.defaultExecutorType = defaultExecutorType;
+    this.defaultExecutorType = defaultExecutorType == null ? ExecutorType.SIMPLE : defaultExecutorType;
```

Comment:
It might be better to keep the current value if the argument is null.
```java
if (defaultExecutorType != null) {
  this.defaultExecutorType = defaultExecutorType;
}
```